### PR TITLE
Add component examples density and mode controls

### PR DESCRIPTION
**Pull request by JoanaMMoreira**
- Add a wrapper `LivePreviewControls` that provides density and mode controls to component examples via context (list view will be added on a separate PR)
- Preview: https://saltdesignsystem-git-add-live-preview-controls-joshwooding.vercel.app/salt/components/button/examples

### CODE REVIEW

File: site/src/components/components/LivePreview.module.css [hunk 1]
```diff
@@ -25,6 +25,11 @@
   align-items: center;
   justify-content: center;
   gap: calc(2 * var(--salt-size-unit));
+  flex-wrap: wrap;
+}
+
+.darkMode {
+  background-color: var(--salt-color-gray-900);
```
Comment (joshwooding) [inline site/src/components/components/LivePreview.module.css:32]: This looks pretty fragile. Is there a way to have the Provider wrap this and set the background?

File: site/src/components/components/LivePreviewControls.module.css [hunk 2]
```diff
@@ -0,0 +1,31 @@
+.controls {
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: space-between;
+  padding: calc(var(--salt-size-unit) * 2) 0;
+  border-bottom: var(--salt-size-border) var(--salt-separable-borderStyle) var(--salt-separable-tertiary-borderColor);
+  background-color: var(--salt-palette-accent-foreground);
```
Comment (libertymayc) [inline site/src/components/components/LivePreviewControls.module.css:7]: is foreground wrong here? and it should be --salt-accent used

Comment (libertymayc) [reply site/src/components/components/LivePreviewControls.module.css:7]: i think it's probably --salt-container-primary-bg here

File: site/src/components/components/LivePreviewControls.module.css [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  position: sticky;
+  top: var(--navbar-height);
+  z-index: var(--salt-zIndex-default);
+  gap: calc(var(--salt-size-unit) * 3);
+}
+
+.toggleButtonGroups {
+  display: flex;
+  flex-wrap: wrap;
+  gap: calc(var(--salt-size-unit) * 3);
```
[the remark this answers is not in the document]
Comment (JoanaMMoreira) [reply site/src/components/components/LivePreviewControls.module.css:17]: We can only make this change after the next mosaic release as these vars are not currently available unfortunately

**Comment by JoanaMMoreira** [comment]
@libertymayc the tooltip issue should also be fixed after the next Mosaic release

**Review by libertymayc — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Pull request merged**